### DISCUSSION TITLE
DataSync.Client and WinUI 3

### DISCUSSION THREAD

**Discussion by CaseNewmark** [Azure/azure-mobile-apps#367] [Q&A] [answered]
Hi,
just yesterday I ported my WPF tech demo to WinUi 3 and whereas DataSync.Client 5.0.3 works well in the WPF scenario (based on your nice tutorials), it doesn't for the WinUI project.
It fails saying that it isn't able to load the Microsoft.MAUI.essentials during runtime when newing DatasyncClient.
The main difference between the projects is the TargetFramework net6.0-windows for WPF and net6.0-windows10.0.19041.0 for WinUi.
Any thoughts on this?
BR,
Daniel

Btw downgrading is also not working. It changes to a problem with a e_sqlite3.dll

**Comment by adrianhall** [comment, marked as answer]
I have never used a WinUI project.  I suspect that our dependency on MAUI is a problem here.

Can you write a "TodoApp" similar to the sample apps that are already there, using WinUI3 and the InMemoryTodoService (instead of the RemoteTodoService).  Once I have that, I can take a look.  

Also, file a feature request for WinUI3 support.

**Reply by CaseNewmark** [reply]
Sorry, we had a bank holiday yesterday. I did a Winui TodoApp and can reproduce the problem if I use the RemoteTodoService. I can't push, so tell how I should send you the branch.

**Reply by adrianhall** [reply]
Can you do a PR that introduces a TodoApp.WinUI3 (in the windows folder) into the TodoApp sample?

**Reply by adrianhall** [reply]
... or just invite me to the sample repo and I will copy the code over to the sample this weekend.
